### PR TITLE
Update the description of the isolated CPUs field

### PR DESCRIPTION
**Pull request by cynepco3hahue**
Add notice regarding using isolated CPUs as complementary to reserved CPUs.

Signed-off-by: Artyom Lukianov <alukiano@redhat.com>

### CODE REVIEW

**Comment by cynepco3hahue** [comment]
@MarSik FYI

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml [hunk 1]
```diff
@@ -54,6 +54,11 @@ spec:
                   description: Isolated defines a set of CPUs that will be used to
                     give to application threads the most execution time possible,
                     which means removing as many extraneous tasks off a CPU as possible.
+                    It important to notice, that the CPU manager can choose any CPU
```
Comment (yanirq) [inline deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:57]: It -> It is

Comment (fromanirh) [inline deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:57]: I'd say either
`It is important to notice`
or simply:
`Important notice:`

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml [hunk 2]
```diff
@@ -54,6 +54,11 @@ spec:
...
+                    to run the workload, except reserved CPUs, so to guarantee that
```
Comment (yanirq) [inline deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:58]: change to :  to run the workload except reserved CPUs. In order to guarantee...

Comment (fromanirh) [inline deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:58]: drop the first comma:
`to run the workload except reserved CPUs,...`

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml [hunk 3]
```diff
@@ -54,6 +54,11 @@ spec:
...
+                    your workload will run on the isolated CPU, reserved and isolated
+                    CPUs should include all online CPUs and isolated CPUs field should
```
Comment (yanirq) [inline deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:60]: online ? or all CPUs ?

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml [hunk 4]
```diff
@@ -54,6 +54,11 @@ spec:
...
+                    be complementary to reserved CPUs.
```
Comment (fromanirh) [inline deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:61]: I'd rephrase slightly like this:
`your workload will run on the isolated CPU: 1. the union of reserved CPUs and isolated CPUs should include all online CPUs; 2. the isolated CPUs field should be the complementary to reserved CPUs field.`

File: deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml [hunk 5]
```diff
@@ -54,6 +54,11 @@ spec:
                   description: Isolated defines a set of CPUs that will be used to
                     give to application threads the most execution time possible,
                     which means removing as many extraneous tasks off a CPU as possible.
+                    It important to notice, that the CPU manager can choose any CPU
+                    to run the workload, except reserved CPUs, so to guarantee that
+                    your workload will run on the isolated CPU, reserved and isolated
+                    CPUs should include all online CPUs and isolated CPUs field should
+                    be complementary to reserved CPUs.
```
Comment (fromanirh) [inline deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml:61]: same, copypaste here

File: docs/performance_profile.md [hunk 6]
```diff
@@ -26,7 +26,7 @@ CPU defines a set of CPU related features.
 | Field | Description | Scheme | Required |
 | ----- | ----------- | ------ | -------- |
 | reserved | Reserved defines a set of CPUs that will not be used for any container workloads initiated by kubelet. | *[CPUSet](#cpuset) | false |
-| isolated | Isolated defines a set of CPUs that will be used to give to application threads the most execution time possible, which means removing as many extraneous tasks off a CPU as possible. | *[CPUSet](#cpuset) | false |
+| isolated | Isolated defines a set of CPUs that will be used to give to application threads the most execution time possible, which means removing as many extraneous tasks off a CPU as possible. It important to notice, that the CPU manager can choose any CPU to run the workload, except reserved CPUs, so to guarantee that your workload will run on the isolated CPU, reserved and isolated CPUs should include all online CPUs and isolated CPUs field should be complementary to reserved CPUs. | *[CPUSet](#cpuset) | false |
```
Comment (fromanirh) [inline docs/performance_profile.md:29]: same I guess

File: pkg/apis/performance/v1/performanceprofile_types.go [hunk 7]
```diff
@@ -50,6 +50,10 @@ type CPU struct {
 	Reserved *CPUSet `json:"reserved,omitempty"`
 	// Isolated defines a set of CPUs that will be used to give to application threads the most execution time possible,
 	// which means removing as many extraneous tasks off a CPU as possible.
+	// It important to notice, that the CPU manager can choose any CPU to run the workload,
+	// except reserved CPUs, so to guarantee that your workload will run on the isolated CPU,
+	// reserved and isolated CPUs should include all online CPUs and isolated CPUs
+	// field should be complementary to reserved CPUs.
```
Comment (fromanirh) [inline pkg/apis/performance/v1/performanceprofile_types.go:56]: ditto

File: pkg/apis/performance/v1alpha1/performanceprofile_types.go [hunk 8]
```diff
@@ -50,6 +50,10 @@ type CPU struct {
 	Reserved *CPUSet `json:"reserved,omitempty"`
 	// Isolated defines a set of CPUs that will be used to give to application threads the most execution time possible,
 	// which means removing as many extraneous tasks off a CPU as possible.
+	// It important to notice, that the CPU manager can choose any CPU to run the workload,
+	// except reserved CPUs, so to guarantee that your workload will run on the isolated CPU,
+	// reserved and isolated CPUs should include all online CPUs and isolated CPUs
+	// field should be complementary to reserved CPUs.
```
Comment (fromanirh) [inline pkg/apis/performance/v1alpha1/performanceprofile_types.go:56]: ditto

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml (hunk 3 above)
Comment (fromanirh) [reply deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:60]: Good point. Thinking about SMT support, I'd say "online". AFAIK there is no other case on which `online_cpus != installed_cpus`. Also, we don't expect cpus to be put online/offline during the worker lifecycle, right?
cc @MarSik to doublecheck.

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml (hunk 1 above)
Comment (cynepco3hahue) [reply deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:57]: done

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml (hunk 2 above)
Comment (cynepco3hahue) [reply deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:58]: done

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml (hunk 1 above)
Comment (cynepco3hahue) [reply deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:57]: done

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml (hunk 2 above)
Comment (cynepco3hahue) [reply deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:58]: done

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml (hunk 4 above)
Comment (cynepco3hahue) [reply deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:61]: done

File: deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml [hunk 9]
```diff
@@ -51,9 +51,15 @@ spec:
                     Defaults to "true"
                   type: boolean
                 isolated:
-                  description: Isolated defines a set of CPUs that will be used to
+                  description: 'Isolated defines a set of CPUs that will be used to
                     give to application threads the most execution time possible,
                     which means removing as many extraneous tasks off a CPU as possible.
+                    It is important to notice, that the CPU manager can choose any
```
Comment (marcel-apf) [inline deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml:57]: I would reword (not that my English is that good!):
"It is important to notice the CPU manager..."

File: deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml [hunk 10]
```diff
@@ -51,9 +51,15 @@ spec:
...
+                    CPU to run the workload except reserved CPUs. In order to guarantee
```
Comment (marcel-apf) [inline deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml:58]: except "the" reserved CPUs

File: deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml [hunk 11]
```diff
@@ -51,9 +51,15 @@ spec:
...
+                    that your workload will run on the isolated CPU:   1. the union
```
Comment (marcel-apf) [inline deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml:59]: The union? (capital letter)

File: deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml [hunk 12]
```diff
@@ -51,9 +51,15 @@ spec:
...
+                    of reserved CPUs and isolated CPUs should include all online CPUs   2.
+                    the isolated CPUs field should be the complementary to reserved
```
Comment (marcel-apf) [inline deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml:61]: The isolated.. (capital letters)

File: docs/performance_profile.md [hunk 13]
```diff
@@ -26,7 +26,7 @@ CPU defines a set of CPU related features.
 | Field | Description | Scheme | Required |
 | ----- | ----------- | ------ | -------- |
 | reserved | Reserved defines a set of CPUs that will not be used for any container workloads initiated by kubelet. | *[CPUSet](#cpuset) | false |
-| isolated | Isolated defines a set of CPUs that will be used to give to application threads the most execution time possible, which means removing as many extraneous tasks off a CPU as possible. | *[CPUSet](#cpuset) | false |
+| isolated | Isolated defines a set of CPUs that will be used to give to application threads the most execution time possible, which means removing as many extraneous tasks off a CPU as possible. It is important to notice, that the CPU manager can choose any CPU to run the workload except reserved CPUs. In order to guarantee that your workload will run on the isolated CPU:\n  1. the union of reserved CPUs and isolated CPUs should include all online CPUs\n  2. the isolated CPUs field should be the complementary to reserved CPUs field | *[CPUSet](#cpuset) | false |
```
Comment (marcel-apf) [inline docs/performance_profile.md:29]: the same ....  to notice the CPU manager ....
and the same on capital letters

File: deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml (hunk 9 above)
Comment (cynepco3hahue) [reply deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml:57]: done

File: deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml (hunk 10 above)
Comment (cynepco3hahue) [reply deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml:58]: done
